scripts/generate_autolink_android.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+function generate_android_autolink() {
+  AUTOLINKING_DIR="android/build/generated/autolinking"
+
+  if [ ! -d "$AUTOLINKING_DIR" ]; then
+      mkdir -p "$AUTOLINKING_DIR"
+      echo "Created directory: $AUTOLINKING_DIR"
+  fi
+
+  rm "$AUTOLINKING_DIR/autolinking.json" || true
+  react-native config > "$AUTOLINKING_DIR/autolinking.json"
+  echo "Generated autolinking.json"
+
+  ROOT_VALUE=$(jq -r '.root' "$AUTOLINKING_DIR/autolinking.json")
```

Comment:
Not a single comment explaining what this is and why we need to do this.

Reply:
The issue is that `autolinking.json` generated contains a full static path to the dependencies in `node_modules` and we face issues when android nix derivation tries to access those paths.

In this script i take the root key which is present in this json and which also contains the path to my project. similar to $GIT_ROOT and I replace that with ".." to make sure that the path becomes a relative path to `node_modules` directory.

I'll add a comment and explain why I had to do this.